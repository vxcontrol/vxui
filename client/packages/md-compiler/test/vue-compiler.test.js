const compile = require('../lib/vue-compiler')

function snapshot (title, def) {
  test(title, () => {
    expect(compile(def)).toMatchSnapshot()
  })
}

snapshot('basic', {
  template: '<div>content</div>'
})

snapshot('data', {
  data: {
    foo: 'bar'
  },
  template: '<div>content</div>'
})

snapshot('components', {
  components: {
    foo: {
      template: '<div>component foo</div>'
    },
    bar: {
      data: {
        foo: 'bar'
      },
      template: '<div>component bar</div>'
    }
  },
  template: '<div>content</div>'
})
