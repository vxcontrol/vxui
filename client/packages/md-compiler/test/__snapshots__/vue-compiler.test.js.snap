// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"{
  render: function render() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c('div', [_vm._v('content')])
  }
}"
`;

exports[`components 1`] = `
"{
  render: function render() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c('div', [_vm._v('content')])
  },
  components: {
    foo: {
      render: function render() {
        var _vm = this
        var _h = _vm.$createElement
        var _c = _vm._self._c || _h
        return _c('div', [_vm._v('component foo')])
      }
    },
    bar: {
      render: function render() {
        var _vm = this
        var _h = _vm.$createElement
        var _c = _vm._self._c || _h
        return _c('div', [_vm._v('component bar')])
      },
      data: function() {
        return { foo: 'bar' }
      }
    }
  }
}"
`;

exports[`data 1`] = `
"{
  render: function render() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c('div', [_vm._v('content')])
  },
  data: function() {
    return { foo: 'bar' }
  }
}"
`;
